vm/devices/get/guest_emulation_device/src/lib.rs

```diff
@@ -618,20 +621,26 @@ impl<T: RingMem + Unpin> GedChannel<T> {
     }
 
     fn handle_time(&mut self) -> Result<(), Error> {
-        const WINDOWS_EPOCH: time::OffsetDateTime = time::macros::datetime!(1601-01-01 0:00 UTC);
+        const WINDOWS_EPOCH: DateTime = date(1601, 1, 1).at(0, 0, 0, 0);
 
         // utc in TimeResponse is in units of 100ns since the windows epoch
-        let now_utc = time::OffsetDateTime::now_utc();
-        let since_win_epoch = now_utc - WINDOWS_EPOCH;
-        let since_win_epoch: i64 = (since_win_epoch.whole_nanoseconds() / 100)
-            .try_into()
-            .unwrap();
-
-        // time_zone is in minutes between UTC and local time (as stored
+        let since_win_epoch = (WINDOWS_EPOCH
+            .to_zoned(TimeZone::UTC)
+            .expect("windows epoch value to be valid")
+            .timestamp()
+            .duration_until(Timestamp::now())
+            .as_nanos()
+            / 100) as i64;
+
+        // tz_offset is in minutes between UTC and local time (as stored
         // in a windows TIME_ZONE_INFORMATION struct)
-        let local_offset = time::UtcOffset::current_local_offset().unwrap_or(time::UtcOffset::UTC);
-        let time_zone = local_offset.whole_minutes();
-        let response = get_protocol::TimeResponse::new(0, since_win_epoch, time_zone, false);
+        let local_offset = TimeZone::try_system().unwrap_or(TimeZone::UTC);
+        let tz_offset = (local_offset
+            .to_fixed_offset()
+            .expect("the system timezone to be fixed")
+            .seconds()
+            / 60) as i16;
```

Comment:
So I think what you're trying to do is get the _local_ offset at a _particular_ time. For that, you just want `Zoned::now().offset()`.

So I'd start this function with `let now = Zoned::now();`. Then you can use `now.timestamp()` in the call to `duration_until` above. And then `now.offset().seconds()` to get the actual offset.

`to_fixed_offset` is specifically only going to work if you _know_ your `TimeZone` is just a fixed offset. It is basically never correct to call this and `unwrap()` it on a time zone retrieved from the system. (It'd be rare for it to even work I imagine.)

Reply:
Thanks, this is absolutely what I had intended. I tried to do a translation from the `time` functions to `jiff` but didn't read deep enough into what a fixed offset was. As an added bonus, the code is much simpler now!